### PR TITLE
Added more troubleshooting instructions

### PR DESCRIPTION
**Pull request by saturnism**
For #4103

### CODE REVIEW

File: SECURITY.md [hunk 1]
```diff
@@ -318,13 +318,16 @@ public <ReqT, RespT> ServerCall.Listener<ReqT> interceptCall(ServerCall<RespT> c
...
- - or that a wrong version is used due to dependency management.
+ - or that a wrong version is used due to dependency management
+ - or you are on an unsupported platform (e.g., 32-bit OS, Alpine with `muslc`). See [Transport Security](#transport-security-tls) for supported platforms.
 
 ### Netty
 If you aren't using gRPC on Android devices, you are most likely using `grpc-netty` transport.
 
 If you are developing for Android and have a dependency on `grpc-netty`, you should remove it as `grpc-netty` is unsupported on Android. Use `grpc-okhttp` instead.
 
+If you are on a 32-bit operating system, or not on an [Transport Security supported platform](#transport-security-tls), you should use Jetty ALPN (and beware of potential issues), or you'll need to build your own 32-bit version of `netty-tcnative`.
```
Comment (carl-mastrangelo) [inline SECURITY.md]: s/on an/on a/

File: SECURITY.md [hunk 2]
```diff
@@ -337,12 +340,16 @@ If you have both `netty-handler` and `netty-tcnative-boringssl-static` dependenc
...
-If you are using `musl` libc (e.g., with Alpine Linux), then `netty-tcnative-boringssl-static` won't work. There are several alternatives:
- - Use [netty-tcnative-alpine] (https://github.com/pires/netty-tcnative-alpine)
- - Use a distribution with `glibc`
+If you are running in a runtime environment that also uses Netty (e.g., Hadoop, Spark, Spring Boot 2) and you have no control over the Netty version at all, then you should use a shaded gRPC Netty dependency to avoid classpath conflicts with other Netty versions in runtime the classpath:
+ - Remove `io.grpc:grpc-netty` dependency
+ - Add `io.grpc:grpc-netty-shaded` dependency
 
 If you are running inside of an embedded Tomcat runtime (e.g., Spring Boot), then some versions of `netty-tcnative-boringssl-static` will have conflicts and won't work. You must use gRPC 1.4.0 or later.
 
+If you are using `musl` libc (e.g., with Alpine Linux), then `netty-tcnative-boringssl-static` won't work. There are several alternatives:
```
Comment (carl-mastrangelo) [inline SECURITY.md]: should this be `musl` or `muslc` as above?

**Comment by ejona86** [comment]
@saturnism, does this mean you want this change over #4185?

File: SECURITY.md (hunk 1 above)
Comment (saturnism) [reply SECURITY.md]: fixed

File: SECURITY.md (hunk 2 above)
Comment (saturnism) [reply SECURITY.md]: oops, i think i'll use the same musl libc wording. good catch!

**Pull request merged**